cai_causal_graph/identify_utils.py

```diff
@@ -13,14 +13,44 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 """
-from typing import List
+from typing import List, Tuple
 
 from cai_causal_graph import CausalGraph
 from cai_causal_graph.exceptions import CausalGraphErrors
 from cai_causal_graph.graph_components import Node
 from cai_causal_graph.type_definitions import NodeLike
 
 
+def _verify_identify_inputs(graph: CausalGraph, node_1: NodeLike, node_2: NodeLike) -> Tuple[str, str]:
+    """
+    Verify the inputs to the identify utilities.
+
+    :param graph: The causal graph given by a `cai_causal_graph.causal_graph.CausalGraph` instance. This must be a DAG,
+        i.e. it must only contain directed edges and be acyclic, otherwise a `TypeError` is raised.
+    :param node_1: The first node or its identifier.
+    :param node_2: The second node or its identifier.
```

Comment:
Would mention these need to be unique. As we raise `ValueError` when not.

Reply:
I'm not convinced by this. Yes, we do raise a `ValueError` but if they try to put the same node for `node_1` and `node_2` they are missing the point of the function, i.e. finding the set of confounders between a node and itself does not seem a sensible thing to do. If there is an edge where someone would try this for whatever reason, we already do have the `ValueError` to notify them.

Reply:
We tell them we raise `TypeError`. Was more just saying as a means to expect a certain exception type. It is moot as this is protected anyways so won't be shown in docs. More just a message for dev. But we don't say about the `NodeDoesNotExistError` either so will just mark as resolved. No action needed.